src/dex/uniswap.js

```diff
@@ -33,23 +45,77 @@ export default class UniswapAdaptor {
     };
     try {
       const blockNumber = await this._provider.getBlockNumber();
-      if (blockNumber === this._lastBlock) return;
-      this._lastBlock = blockNumber;
 
-      const offer = await this._uniswap.getAmountsOut(
-        ilkAmount, Config.vars.collateral[this._collateralName].uniswapRoute
-      );
-      book.sellAmount = ethers.utils.formatUnits(
-        offer[0].mul(this._decNormalized)
-      );
-      book.receiveAmount = ethers.utils.formatUnits(
-        offer[offer.length - 1]
-      );
-      return book;
+      if (typeof(Config.vars.collateral[this._collateralName].token0) !== 'undefined') {
+        const token = new ethers.Contract(
+          Config.vars.collateral[this._collateralName].erc20addr,
+          UniswapV2Pair,
+          this._provider
+        );
+        const totalSupply = await token.totalSupply();
+        const share = ilkAmount.mul(decimals18).div(totalSupply);
+        const reserves = await token.getReserves();
+
+        // normalize reserves for each tokens decimals
+        let ilkAmount0 = reserves[0].mul(share).div(decimals18);
+        let ilkAmount1 = reserves[1].mul(share).div(decimals18);
+        // debug
+        // let ilkAmount0 = BigNumber.from('10').pow(24); // 1 million
+        // let ilkAmount1 = BigNumber.from('10').pow(24); // 1 million
+        // console.log(
+        //   'ilkAmount0: ' +
+        //   ethers.utils.formatUnits(
+        //     ilkAmount0,
+        //     Config.vars.collateral[this._collateralName].token0.decimals
+        //   )
+        // );
+        // console.log(
+        //   'ilkAmount1: ' +
+        //   ethers.utils.formatUnits(
+        //     ilkAmount1,
+        //     Config.vars.collateral[this._collateralName].token1.decimals
+        //   )
+        // );
+
+        let offer0 = [ilkAmount0];
+        let offer1 = [ilkAmount1];
+
+        if (Config.vars.collateral[this._collateralName].token0.name !== 'DAI') {
+          offer0 = await this._uniswap.getAmountsOut(
+            ilkAmount0,
+            Config.vars.collateral[this._collateralName].token0.route
+          );
+        }
+        if (Config.vars.collateral[this._collateralName].token0.name !== 'DAI') {
```

Comment:
Shouldn't this conditional be checking whether `token1` is DAI?

Reply:
yes, fixing